File: 2021/Day 7/TS/part1.ts
import fs from 'fs'
import path from 'path'

const file: string = fs.readFileSync(path.resolve(__dirname, '../data.txt'), 'utf8')
const positions = file.split(',').map((str) => parseInt(str))

let fuel = 999999999
let i: number
for (i = 0; true; i++) {
    let newFuel = positions.reduce((sum, pos) => sum + Math.abs(pos - i), 0)
    if (newFuel > fuel) break
    fuel = newFuel
}

console.log('Fuel consumption:', fuel)

File: 2021/Day 07/C++/part1.cpp
// Headers
#include <algorithm>
#include <array>
#include <cmath>
#include <fstream>
#include <iostream>
#include <map>
#include <sstream>
#include <vector>

// Namespaces
using namespace std;  // Bad practice, I know

// Prototypes

// Main
int main() {
    // Get input
    ifstream file("../data.txt");
    vector<int> positions;
    int position;
    while (file >> position) {
        positions.push_back(position);
        file.ignore();
    }

    // Calculate fuel consumption
    int fuel = numeric_limits<int>::max();
    for (int i = 0; true; i++) {
        int newFuel = 0;
        for (int pos : positions) newFuel += abs(pos - i);
        if (newFuel > fuel) break;
        fuel = newFuel;
    }

    // Output result
    cout << "Fuel consumption: " << fuel << endl;

    return 0;
}